cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -567,6 +582,42 @@ impl ABIMachineSpec for X64ABIMachineSpec {
         insts
     }
 
+    fn gen_memcpy(
+        call_conv: isa::CallConv,
+        dst: Reg,
+        src: Reg,
+        size: usize,
+    ) -> SmallVec<[Self::I; 8]> {
+        let mut insts = SmallVec::new();
```

Comment:
Here we should probably use the `emit_vm_call`.

Reply:
So I started down this path, but pulling on the string actually unravels a *lot* of things -- `emit_vm_call` takes a LowerCtx and emits directly, which we need to do because the caller-side ABI object that it uses requires this; but `gen_memcpy()` returns a list of instructions because we cannot parameterize on the `LowerCtx` in the callee-side ABI object, because it is used as a dyn trait. This is slightly unfortunate, to say the least. If you think it will be a problem in practice then I can do this, but it requires altering the caller-side stuff to use more Vecs. In practice I *think* this will be OK because: struct args will only show up in SystemV and Fastcall cases; in both cases we can handle three register args; no stack adjustment or other ABI stuff should be necessary. Thoughts?

Reply:
Ah, didn't see the LowerCtx trait coming! Okay, I'm convinced that this shouldn't be a problem for Fastcall too, so we can keep this code like this. If there's a `call_conv` hanging around, could we assert it's either equal to `SystemV` or `Fastcall`, just for future-proofing?